src/ngx_http_lua_socket_tcp.c

```diff
@@ -2784,7 +2811,13 @@ ngx_http_lua_socket_tcp_settimeouts(lua_State *L)
         }
     }
 
-    return 0;
+    lua_pushboolean(L, 1);
+    return 1;
+
+timeout_overflow:
+    lua_pushnil(L);
+    lua_pushliteral(L, "timeout overflow");
```

Comment:
Again, we should throw out a Lua exception for this since it's a programming mistake.